### DISCUSSION TITLE
방문자 리스트 csv 파일 추출하기

### DISCUSSION THREAD

**Discussion by Jsueeee** [euphony-io/check-in-listener#19] [General]
@euphony-io/eu-us 

#18 수진님이 만들어주신 csv 파일 추출에 대해서 코드리뷰 해주세요!
수정해야할 부분에 대해서도 토론했으면 좋겠습니다!

**Comment by Jsueeee** [comment]
@sujin9 csv 파일 추출 관련해서 참고하신 자료 링크 공유해주실 수 있을까요?
팀원 모두 공부했으면 좋겠습니다! @euphony-io/eu-us

**Comment by sujin9** [comment]
구글에 android export csv , android export room data to csv 이런 식으로 검색해서 많이 이것저것 찾아봤었어요!

https://mobikul.com/room-database-export-import-to-csv/
https://medium.com/@tonia.tkachuk/how-to-export-room-database-to-csv-file-a51fec589618

**Comment by sujin9** [comment]
아마 지금 PR에 올려진 코드는 문제가 있을 거 같아요! 데이터베이스가 null이어서, CSV 파일 생성까지는 되는데 내용을 못불러와요..!
그래서 [링크](https://stackoverflow.com/questions/51451819/how-to-get-context-in-android-mvvm-viewmodel)를 참고해서 코드를 수정하고 데이터베이스 초기화를 했더니,
> java.lang.IllegalStateException: Cannot access database on the main thread since it may potentially lock the UI for a long period of time.

이런 오류가 ㅎㅎ....
